Modules/Challenges/src/Challenges.Infrastructure/Persistence/Database/Repository/ChallengesRepository.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Backbone.Modules.Challenges.Application.Extensions;
+using Backbone.Modules.Challenges.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Challenges.Domain.Entities;
+using Backbone.Modules.Challenges.Domain.Ids;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backbone.Modules.Challenges.Infrastructure.Persistence.Database.Repository;
+public class ChallengesRepository : IChallengesRepository
+{
+    private readonly DbSet<Challenge> _challenges;
+    private readonly ChallengesDbContext _dbContext;
+    
+    public ChallengesRepository(ChallengesDbContext dbContext)
+    {
+        _challenges = dbContext.Challenges;
+        _dbContext = dbContext;
+    }
+    public async Task<Challenge> Find(ChallengeId id, CancellationToken cancellationToken)
+    {
+        return await _challenges
+            .NotExpired()
```

Comment:
I think you should always return the Challenge, whether it is expired or not. In the GetChallengeByIdCommand Handler, you should then check if the Challenge is expired. Do this by adding an `IsExpired` method to the `Challenge` class. If the Challenge is expired, throw a NotFoundException.
The disadvantage of this approach is, that the database needs to return the data of the Challenge even though it isn't needed. But the advantage is, that the logic of "don't return the Challenge if it is expired" stays in the Application logic, and is not moved to the Infrastructure part.